wgpu-hal/src/lib.rs

```diff
@@ -558,6 +558,7 @@ pub trait Adapter: WasmNotSendSync {
         &self,
         features: wgt::Features,
         limits: &wgt::Limits,
+        memory_hints: &wgt::MemoryHints,
```

Comment:
taking a reference to `MemoryHints` seems a lil bit silly to me. Why not make it `Copy` and pass by value?

Reply:
My thinking was that I'm not entirely sure how the parameters will evolve. If you feel strongly about it, I don't mind passing them by value and even make them copy. We don't have a commitment to not break the API.

Reply:
🤷 was mostly surprised. Not feeling strongly about either direction.